### PR TITLE
feat(storage): Use drop_table_by_id api in `drop all`

### LINKED ISSUE #5626: Feat: catalog.drop_table_by_id
> Is there a consistency issue that `drop_table` at line 79 and `truncate()` at line 86 will take place on different tables?
> 
> _Originally posted by @drmingdrmer in https://github.com/datafuselabs/databend/pull/5619#discussion_r883231426_

------------------

- [ ] introduce `drop_table_by_id` method to catalog
    which ... drops a table by id :)
- [ ] refine the impl of `drop all`
    use `drop_table_byd_id` instread of drop table -> 
https://github.com/datafuselabs/databend/blob/797c1acb205ea7cdcdc37d5adabf97139580a135/query/src/interpreters/interpreter_table_drop.rs#L79
    just follow the `tbl.get_table_info`, and find the table's id

### PR DESCRIPTION
**Pull request by johnhaxx7**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Use `drop_table_by_id` introduced in https://github.com/datafuselabs/databend/pull/9990, to reduce the chance of duplicate table name race condition 

No new test added, it's covered by `tests/sqllogictests/suites/base/05_ddl/05_0001_ddl_drop_table_full`

Closes #5626

### CODE REVIEW

File: src/query/service/src/interpreters/interpreter_table_drop.rs [hunk 1]
```diff
@@ -73,6 +72,11 @@ impl Interpreter for DropTableInterpreter {
                 let latest = tbl.as_ref().refresh(self.ctx.as_ref()).await?;
                 latest.truncate(self.ctx.clone(), purge).await?
             }
+        } else {
+            return Err(ErrorCode::TableInfoError(format!(
```
Comment (johnhaxx7) [inline src/query/service/src/interpreters/interpreter_table_drop.rs:76]: Hi @BohuTANG, I added this check because I think this branch in if condition is error prone. We don't know what will happen if Option<tbl> is empty, we just return  `Ok(PipelineBuildResult::create())`, which IMHO can be hard to debug in some edge cases. After I add this, a series of tests are failing, like https://github.com/datafuselabs/databend/actions/runs/4181126745/jobs/7242889308, can you pull in some one to explain a bit on, in which case tbl is empty? So that I can add some comment here to help explain the scenario.

**Comment by johnhaxx7** [comment]
@drmingdrmer PTAL, thanks!

**Review by drmingdrmer — COMMENTED** [review]
The codes looks good! Thank you!
Although, personally speaking, I'd not suggest adding more methods to the catalog. Deleting a table is quite a straight forward operation, thus get a handle to meta-service from context and then just call a meta-client method(SchemaApi) would be simple and leader.

**Comment by BohuTANG** [comment]
> The codes looks good! Thank you! Although, personally speaking, I'd not suggest adding more methods to the catalog. Deleting a table is quite a straight forward operation, thus get a handle to meta-service from context and then just call a meta-client method(SchemaApi) would be simple and leader.

Agree.
Change the `drop_table(plan)` to `drop_table(table_id)`? @drmingdrmer

**Comment by drmingdrmer** [comment]
> Agree. Change the `drop_table(plan)` to `drop_table(table_id)`? @drmingdrmer

Sounds good!

File: src/query/service/src/catalogs/default/database_catalog.rs [hunk 2]
```diff
@@ -325,25 +325,7 @@ impl Catalog for DatabaseCatalog {
...
-            .await?
-        {
-            return self.immutable_catalog.drop_table(req).await;
-        }
-        self.mutable_catalog.drop_table(req).await
-    }
-
-    async fn drop_table_by_id(&self, tb_id: MetaId) -> Result<DropTableReply> {
-        let res = self.mutable_catalog.drop_table_by_id(tb_id).await?;
+        let res = self.mutable_catalog.drop_table(req).await?;
```
Comment (drmingdrmer) [inline src/query/service/src/catalogs/default/database_catalog.rs:328]: The method name would be better to be explicit about what it exactly does, by naming it `drop_table_by_id`

**Review by drmingdrmer — COMMENTED** [review]
Glad to see codes being removed!

File: src/query/service/src/catalogs/default/database_catalog.rs (hunk 2 above)
Comment (johnhaxx7) [reply src/query/service/src/catalogs/default/database_catalog.rs]: updated to `drop_table_by_id`, PTAL, thanks!

File: src/meta/api/src/schema_api.rs [hunk 3]
```diff
@@ -102,7 +102,7 @@ pub trait SchemaApi: Send + Sync {
     /// It returns a list of (table-id, table-meta-seq, table-meta).
     async fn list_all_tables(&self) -> Result<Vec<(TableId, u64, TableMeta)>, KVAppError>;
 
-    async fn drop_table(&self, req: DropTableReq) -> Result<DropTableReply, KVAppError>;
+    async fn drop_table_by_id(&self, req: DropTableReq) -> Result<DropTableReply, KVAppError>;
```
Comment (drmingdrmer) [inline src/meta/api/src/schema_api.rs:105]: The same convention applies to `DropTableReq -> DropTableByIdReq`.

**Comment by BohuTANG** [comment]
`make lint` in your local will make the `Dev Linux / check` happy:)

**Review by drmingdrmer — APPROVED** [review]

**Review by BohuTANG — APPROVED** [review]
👍

**Comment by johnhaxx7** [comment]
@mergifyio refresh

**Comment by johnhaxx7** [comment]
@Mergifyio refresh
